testing/srnca/test_nca.py

```diff
@@ -53,14 +53,74 @@ def test_save_parameters(self):
 
         os.system(my_command)
 
+    def test_to_device(self):
+
+        nca = NCA()
+        nca.to_device("cpu")
+        nca.to_device("cuda")
+        
+
     def test_fit(self):
 
         nca = NCA(number_channels=3)
 
         target = torch.rand(1,3,64,64)
 
-        nca.fit(target, max_steps=3, lr=1e-3, max_ca_steps=16, batch_size=4)
+        this_filepath = os.path.realpath(__file__)
+        temp_tag = os.path.split(this_filepath)[0]
+
+        temp_tag = os.path.join(temp_tag, "temp")
+        nca.fit(target, max_steps=3, lr=1e-3, max_ca_steps=16, batch_size=4, exp_tag=temp_tag)
+
+
+        find_log = False
+        find_params = False
+        for elem in os.listdir(os.path.split(this_filepath)[0]):
+            if elem.startswith("temp") and elem.endswith(".npy"):
+                find_log = True
+            if elem.startswith("temp") and elem.endswith(".pt"):
+                find_params = True
+
+        self.assertTrue(find_log)
+        self.assertTrue(find_params)
+        os.system(f"rm {os.path.split(this_filepath)[0]}/temp*_log_dict.npy")
+        os.system(f"rm {os.path.split(this_filepath)[0]}/temp*.pt")
+
+
+    def test_command_line(self):
```

Comment:
Note that code executed as if from the command line (_e.g._ `os.system(my_cmd)`) does not get counted by coverage when determining line coverage. I use `#pragma: no cover` on `if __name__ == "__main__":` scopes, so we don't get reports about testing these command-line entry points. 

It's still important to test them, it's not as obvious if we don't (due to the pragma), and the code they call also needs to be tested elsewhere, not from a command line call (to ensure line coverage).